clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1327,16 +1327,25 @@ private void recordPartitionLag(TopicPartition tp, long lag) {
             String name = partitionLagMetricName(tp);
             Sensor recordsLag = this.metrics.getSensor(name);
             if (recordsLag == null) {
+                Map<String, String> metricTags = new HashMap<>(2);
+                metricTags.put("topic", tp.topic().replace('.', '_'));
```

Comment:
Is there a reason you replace '.' by '_' in that case?

Reply:
this is always done in metrics. I believe it's because JMX uses `.` to define hierarchy on entities so using a dot here would cause issues in the jmx reporter.
It's mentioned here: http://kafka.apache.org/documentation.html#upgrade_9_breaking

Reply:
Thanks for your reply. I guess this may be true in some cases, but at least for e.g. `kafka.server:type=BrokerTopicMetrics,name=MessagesInPerSec` it works even if the topic contains dots.